benchmarks/tpch/run.py

```
@@ -0,0 +1,355 @@
+    limit_bytes = memory_limit_mb * 1024 * 1024
+    resource.setrlimit(resource.RLIMIT_AS, (limit_bytes, limit_bytes))
+
+
+def _engine_worker(engine_sql, db_path, source_name, memory_limit_mb, result_queue):
+    """Child-process entry: build a fresh runtime, run one query, send it back."""
+    _apply_memory_limit(memory_limit_mb)
+    try:
+        runtime = build_runtime(db_path, source_name)
+        result = runtime.execute(engine_sql)
+        result_queue.put(("ok", arrow_to_rows(result)))
+    except Exception as error:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)